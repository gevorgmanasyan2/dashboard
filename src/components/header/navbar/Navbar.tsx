import { DownOutlined, SendOutlined } from '@ant-design/icons'
import { Dropdown, Menu, Tooltip } from 'antd'
import React, { FC, MouseEvent, useState } from 'react'
import { Link, NavLink } from 'react-router-dom'
import { baseAPIURL } from './../../../api/constants'
import { useSelector } from 'react-redux'
import { RootState } from '../../../redux/rootReducer'

import './navbar.scss'

interface NavbarProps {
  onClick?: (event: MouseEvent<HTMLElement, MouseEvent>) => void
}

const Navbar: FC<NavbarProps> = () => {
  const [isClicked, setIsClicked] = useState<boolean>(false)
  const checkUserRole = useSelector((state: RootState) => state.auth.sessionToken?.roles)

  const handleDownIconClick = (e: any) => {
    e.preventDefault()
    setIsClicked(!isClicked)
  }

  const menu = (
    <Menu>
      <Menu.Item key="1">
        <Link to="/regulatory-permissions">Regulatory Permissions</Link>
      </Menu.Item>
      <Menu.Item key="2">
        <Link to="/regulatory-library">Resource Library</Link>
      </Menu.Item>
    </Menu>
  )

  return (
    <div className="navbar">
      <NavLink to="/" activeClassName="navbar__link--selected" className="navbar__link">
        <span className="navbar__link--item">My Dashboard</span>
      </NavLink>
      <NavLink to="/regulations" activeClassName="navbar__link--selected" className="navbar__link">
        <Dropdown overlay={menu} trigger={['click']}>
          <span className="ant-dropdown-link" onClick={handleDownIconClick}>
            <span className="navbar__link--item">Regulations</span>
            <DownOutlined
              onClick={handleDownIconClick}
              className={isClicked ? 'navbar__link--itemBackRotate' : 'navbar__link--itemRotate'}
            />
          </span>
        </Dropdown>
      </NavLink>
      {checkUserRole?.includes('device_user') || checkUserRole?.includes('customer_admin') ? (
        <NavLink
          to="route"
          onClick={(event) => {
            event.preventDefault()
            window.location.href = `${baseAPIURL}`
          }}
          activeClassName="navbar__link--selected"
          className="navbar__link"
        >
          <span className="navbar__link--item">
            <Tooltip title='Access FlightDeck to manage your Casia devices and more'>
              FlightDeck <SendOutlined className="sendIcon" />
            </Tooltip>
          </span>
        </NavLink>
      ) : (
        <></>
      )}
    </div>
  )
}

export default Navbar
