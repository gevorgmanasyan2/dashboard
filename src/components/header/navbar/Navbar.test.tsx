/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'
import Navbar from './Navbar'

it('should render Navbar component', () => {
  expect(shallow(<Navbar />)).toMatchSnapshot()
})
