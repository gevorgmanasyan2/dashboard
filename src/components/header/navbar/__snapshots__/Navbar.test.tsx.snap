// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Navbar component 1`] = `
<div
  className="navbar"
>
  <NavLink
    activeClassName="navbar__link--selected"
    className="navbar__link"
    to="/"
  >
    <span
      className="navbar__link--item"
    >
      My Dashboard
    </span>
  </NavLink>
  <NavLink
    activeClassName="navbar__link--selected"
    className="navbar__link"
    to="/regulations"
  >
    <Dropdown
      mouseEnterDelay={0.15}
      mouseLeaveDelay={0.1}
      overlay={
        <Menu>
          <MenuItem>
            <Link
              to="/regulatory-permissions"
            >
              Regulatory Permissions
            </Link>
          </MenuItem>
          <MenuItem>
            <Link
              to="/regulatory-library"
            >
              Resource Library
            </Link>
          </MenuItem>
        </Menu>
      }
      trigger={
        Array [
          "click",
        ]
      }
    >
      <span
        className="ant-dropdown-link"
        onClick={[Function]}
      >
        <span
          className="navbar__link--item"
        >
          Regulations
        </span>
        <ForwardRef(DownOutlined)
          className="navbar__link--itemRotate"
          onClick={[Function]}
        />
      </span>
    </Dropdown>
  </NavLink>
</div>
`;
